### PR TITLE
Remove redundancies with openforcefield

### PR DESCRIPTION
**Pull request by bannanc**
The goal of this pull request is to remove redundancies with the openforcefield repository. 

As of my open pull request #7 on openforcefield, these tests pass on my local machine. I still need to make changes so the travis tests will work on github as well. 
John's suggestion for covering this:
> We could have the smarty travis install the latest development version via
> `pip install git+https://github.com/open-forcefield-group/openforcefield.git`
> if things are going to be changing rapidly within openforcefield for a while.

### CODE REVIEW

**Comment by bannanc** [comment]
@camizanette or @davidlmobley 
If one of your have time to review this PR that would be great. I changed everything to work with openforcefield and removed redundant code with that repository. I didn't remove redundancies in the smarty/data/ directory, should we?

Also - I updated the settings to require a code review.

**Comment by davidlmobley** [comment]
I'll try and review. If there is redundant data between here and `smarty`, I would try to get it removed, yes. Redundancy will be confusing.

**Comment by bannanc** [comment]
@davidlmobley 
Ok, in that case I'll need a little more time to look at the smarty/data/ directory, I'll tag you when I've updated that.

File: examples/README.md [hunk 1]
```diff
@@ -3,9 +3,6 @@
...
-* `SMIRNOFF_simulation/` - gives a simple example of simulating a molecule in the gas phase beginning from a molecular structure and the SMIRNOFF forcefield format.
-* `forcefield_modification/` - gives example of modifying a forcefield parameter, evaluating how it changes an energy (IPython notebook).
-* `chemicalEnvironments/` - contains an example and documentation of using chemical environment objects to manipulate environment being considered, generate example SMIRKS, etc. Also contains IPython notebook using the chemical environment for depiction.
-* `partial_bondorder/` - example of using partial bond orders for parameterizing molecules, and showing how the SMIRFF format extends. (See issue #53 on Smarty).
 * `smirky_simulations/` - example usage of the smirky sampling tool to rediscover the SMIRNOFF99Frosst parameter types
+
+**We have rearranged the Open Force Field group if you are looking for an example that used to be here, but is no longer it can be found at [openforcefield/examples/](https://github.com/open-forcefield-group/openforcefield/tree/master/examples)**
```
Comment (davidlmobley) [inline examples/README.md:8]: Grammar. "We have rearranged the Open Force Field group repositories. If you are looking for an example which used to be here but is no longer, it ..."

**Review by davidlmobley — COMMENTED** [review]
All the changes look good to me so far, @bannanc . Let me know when you're ready for a final review.

**Comment by bannanc** [comment]
@davidlmobley I think this is ready to go now, as far as I know there are no redundancies. 

I did notice/remember that SMIRKY still calls all its variables SMIRFF, but I thought I would write an issue to update that later.

File: smarty/tests/test_atomtyper.py [hunk 2]
```diff
@@ -15,7 +15,7 @@ def test_atomtyper(self):
         typetag = 'atomtype'
         atomtypes = AtomTyper.read_typelist(get_data_filename('atomtypes/basetypes.smarts'))
         replacements = AtomTyper.read_typelist(get_data_filename('atomtypes/replacements.smarts'))
-        molecules = read_molecules(get_data_filename('molecules/zinc-subset-tripos.mol2.gz'), verbose=False)
+        molecules = read_molecules('zinc-subset-tripos.mol2.gz', verbose=False)
```
Comment (davidlmobley) [inline smarty/tests/test_atomtyper.py]: Wait, why does this still pass? Don't you need to direct it to where this is installed, @bannanc ?

File: smarty/tests/test_sampler.py [hunk 3]
```diff
@@ -20,11 +21,11 @@ def __init__(self, *args, **kwargs):
         self.replacements = get_data_filename('atomtypes/replacements.smarts')
 
         # import molecules
-        self.mols_zinc = read_molecules(get_data_filename('molecules/zinc-subset-tripos.mol2.gz'), verbose=False)
-        self.mols_zinc_ref = read_molecules(get_data_filename('molecules/zinc-subset-parm@frosst.mol2.gz'), verbose=False)
+        self.mols_zinc = read_molecules('zinc-subset-tripos.mol2.gz', verbose=False)
```
Comment (davidlmobley) [inline smarty/tests/test_sampler.py]: Likewise, puzzled why you don't need to `get_data_filename` anymore...

**Review by davidlmobley — APPROVED** [review]
Basically looks good except for the two issues noted where I'm confused as to why you've made that change. Can merge once it's addressed/clarified.

File: smarty/tests/test_atomtyper.py (hunk 2 above)
Comment (bannanc) [reply smarty/tests/test_atomtyper.py]: read_molecules uses get_data_filename, I'm not sure why I wrote it that way before.

Comment (bannanc) [reply smarty/tests/test_atomtyper.py]: @davidlmobley 
If the file isn't local read_molecules looks in /openforcefield/data/molecules/

Comment (davidlmobley) [reply smarty/tests/test_atomtyper.py]: Ha, OK, thanks.

**Pull request merged**